pi-manage

```diff
@@ -663,6 +667,37 @@ def rotate_audit(highwatermark=10000, lowwatermark=5000, age=0, config=None,
             print("{0!s} entries deleted.".format(r))
 
 
+@contextlib.contextmanager
+def smartopen(filename):
+    if filename and filename != '-':
+        fh = open(filename, 'w')
+    else:
+        fh = sys.stdout
+
+    try:
+        yield fh
+    finally:
+        if fh is not sys.stdout:
+            fh.close()
+
+
+@audit_manager.command
+def dump(filename):
+    """Dump the audit log into a csv file."""
+    default_module = "privacyidea.lib.auditmodules.sqlaudit"
+    token_db_uri = app.config.get("SQLALCHEMY_DATABASE_URI")
+    audit_db_uri = app.config.get("PI_AUDIT_SQL_URI", token_db_uri)
+    audit_module = app.config.get("PI_AUDIT_MODULE", default_module)
+    if audit_module != default_module:
+        raise Exception("Currently only the SQL audit module is supported. You are using %s" %
+                        audit_module)
```

Comment:
We do not need to read these values from the config file in this case.
In the rotate command, we needed to read this, since the rotating is not implemented in the Audit Module itself.
(It might actually make sense, to put the rotation into the auditmodule one day).

But as in this case, we only need the ``csv_generator`` which is implemented in the auditmodule, this ``dump`` should work for all auditmodules. 

Thus you can drop reading the default module, the config values and raising an exception. (lines 687-693)